cmd/ctl/pkg/install/install.go

```diff
@@ -0,0 +1,230 @@
+a string value use '--set-string'. In case a value is large and therefore
+you want not to use neither '--values' nor '--set', use '--set-file' to read the
+single large value from file.
+`
+
+func NewCmdInstall(ctx context.Context, ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	settings := cli.New()
+	cfg := new(action.Configuration)
+
+	options := &InstallOptions{
+		settings:  settings,
+		cfg:       cfg,
+		client:    action.NewInstall(cfg),
+		valueOpts: &values.Options{},
+
+		IOStreams: ioStreams,
+	}
+
+	// Set default namespace cli flag value
+	defaults := make(map[string]string)
+	defaults["namespace"] = "cert-manager"
+
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "Install cert-manager",
+		Long:  installDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := helm.CopyCliFlags(cmd.Root().PersistentFlags(), defaults, settings); err != nil {
+				return nil
+			}
+			options.client.Namespace = settings.Namespace()
+
+			rel, err := options.runInstall(ctx)
+			if err != nil {
+				return err
+			}
+			return writeRelease(ioStreams.Out, rel, options.DryRun)
+		},
+		SilenceUsage:  true,
+		SilenceErrors: true,
+	}
+
+	addInstallUninstallFlags(cmd.Flags(), &options.client.Timeout, &options.client.Wait)
+	addInstallFlags(cmd.Flags(), options.client)
+	addValueOptionsFlags(cmd.Flags(), options.valueOpts)
+	addChartPathOptionsFlags(cmd.Flags(), &options.client.ChartPathOptions)
+
+	cmd.Flags().BoolVar(&options.client.CreateNamespace, "create-namespace", true, "Create the release namespace if not present")
+	cmd.Flags().StringVar(&options.ChartName, "chart-name", "cert-manager", "Name of the chart to install")
+	cmd.Flags().BoolVar(&options.DryRun, "dry-run", false, "Simulate install and output manifest")
+
+	return cmd
+}
+
+func (o *InstallOptions) runInstall(ctx context.Context) (*release.Release, error) {
+	log.SetFlags(0)
+	log.SetOutput(o.Out)
+
+	if err := o.cfg.Init(o.settings.RESTClientGetter(), o.settings.Namespace(), os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+		return nil, err
+	}
+
+	// Find chart
+	cp, err := o.client.ChartPathOptions.LocateChart(o.ChartName, o.settings)
+	if err != nil {
+		return nil, err
+	}
+
+	chart, err := helm.LoadChart(cp, o.client, o.settings)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if chart is installable
+	if err := helm.CheckIfInstallable(chart); err != nil {
+		return nil, err
+	}
+
+	// Console print if chart is deprecated
+	if chart.Metadata.Deprecated {
+		log.Printf("This chart is deprecated")
+	}
+
+	// Merge all values flags
+	p := getter.All(o.settings)
+	chartValues, err := o.valueOpts.MergeValues(p)
+	if err != nil {
+		return nil, err
+	}
+
+	// Dryrun template generation (used for rendering the crds in /templates)
+	o.client.DryRun = true                  // Do not apply install
+	o.client.IsUpgrade = true               // Do not validate against cluster
```

Comment:
Please explain why setting `IsUpgrade = true` prevents validating against the cluster? It doesn't make sense to me.

Reply:
tThis was a workaround, ideally we use ClientOnly. I figured out how to do that now, so this is not an issue anymore.